### DISCUSSION TITLE
Getting all tunnels for device/vm/interface

### DISCUSSION THREAD

**Discussion by peterbaumert** [netbox-community/netbox#14756] [Help Wanted!] [answered]
Hi all,

I am currently trying to get the tunnels which belong to a device/vm via the api. But I fell like i am having a knot in my brain and can't figure out how to do it ( what queries to do ).

Anyone else tried this already and has some hints?

KR
Pete

**Comment by candlerb** [comment, marked as answer]
I guess you would first get a list of interfaces from the device or VM, and then query for tunnel terminations using interface_id or vminterface_id (which you should be able to repeat for multiple values)

https://github.com/netbox-community/netbox/blob/v3.7.0/netbox/vpn/filtersets.py#L77-L119

**Reply by peterbaumert** [reply]
thanks for untwisting my brain, yeah with (vm)interface_id i can get tunnel_id and with that again all terminations.
